nautobot/extras/jobs.py

```diff
@@ -394,83 +397,108 @@ def class_path(cls):  # pylint: disable=no-self-argument
         """
         return f"{cls.__module__}.{cls.__name__}"
 
+    @final
     @classproperty
-    def class_path_dotted(cls):  # pylint: disable=no-self-argument
+    def class_path_dotted(cls) -> str:  # pylint: disable=no-self-argument
         """
         Dotted class_path, suitable for use in things like Python logger names.
 
         Deprecated as of Nautobot 2.0: just use .class_path instead.
         """
         return cls.class_path
 
+    @final
     @classproperty
-    def class_path_js_escaped(cls):  # pylint: disable=no-self-argument
+    def class_path_js_escaped(cls) -> str:  # pylint: disable=no-self-argument
         """
         Escape various characters so that the class_path can be used as a jQuery selector.
         """
         return cls.class_path.replace(".", r"\.")
 
+    @final
     @classproperty
-    def grouping(cls):  # pylint: disable=no-self-argument
+    def grouping(cls) -> str:  # pylint: disable=no-self-argument
         module = inspect.getmodule(cls)
         return getattr(module, "name", module.__name__)
 
+    @final
+    @classmethod
+    def _get_meta_attr_and_assert_type(cls, attr_name, default, expected_type):
```

Comment:
Perhaps this should be as a mixin or decorator, so if perform type enforcement later it is centralized, well defined, and more easily replaceable?

Reply:
This is solving a fairly narrow problem, that of user-defined code in Jobs not necessarily being conformant to the API for Jobs. I don't think we have any plans now or in the near future to add more generalized type enforcement throughout the code base.